### ISSUE TITLE
3.x broke android access using nabucasa.

### ISSUE THREAD

**Issue by Claudio1L** [AlexxIT/WebRTC#418]
HI

with Nabucasa  and 3.x update, images stuck in loading2 using webrtc-card on Android mobile. Using RTC only. It works with Apple in the same conditions. No significant logs. Just black screen. Going back to 2.3.1 works perfectly.
Following an example of the webrtc card config:

type: custom:webrtc-camera
entity: camera.cam_strada
mode: webrtc
ui: true
background: true

**Comment by Claudio1L** [comment]
2.3.1 seems smoother also on iphone

**Comment by AlexxIT** [comment]
WebRTC on Android is no different from iPhone. You may have a different 4G network configuration. I guess you did not set up external access for WebRTC?

**Comment by Claudio1L** [comment]
ok, thanks.

I read everything related I found, also outside the official docs, but I was not able to solve.
I did some more tests, instead, and I think is something related to my mobile operator. The issue is the same with any iphone and any android. With the operator "HO mobile" works till 2.3.1, not 3.x.  With "TIM" works 3.x also. Consistently.
Any advice on what I can check?

**Comment by PablaV** [comment]
Try removing mode: webrtc from your card config and restart HA and the addon and see if that helps. I am experiencing issues when I set mode: webrtc so that could be the case for you too.

**Comment by Claudio1L** [comment]
Upgraded again to 3.0.2, removed mode: webrtc. no changes, same issue. There is something different between 3.x and 2.x that the mobile operator is filtering. TCP/UDP ports? no idea

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/WebRTC/issues/378

**Comment by Claudio1L** [comment]
> #378

yes thanks, already seen. There is nothing I can use to solve. If it's a simmetric nat issue of the mobile operator, it should work at least with MSE+android, instead nothing. There is something else

**Comment by AlexxIT** [comment]
Check if MSE works for you in LAN. MSE may not work for some buggy cameras.

**Comment by Claudio1L** [comment]
yes

![immagine](https://user-images.githubusercontent.com/97904190/218088676-3a832f1e-0f0f-45d8-8eb1-2eafbe63b94f.png)
![immagine](https://user-images.githubusercontent.com/97904190/218088735-1d8f8485-055a-448c-8957-b50031c277f2.png)

the first one is an hikvision, the second one is a reolink camera

in Lan and via TIM 4G connections everything working, from desktop (firefox+Macos), android and iphone.

**Comment by AlexxIT** [comment]
So can you describe again your issue

**Comment by Claudio1L** [comment]
sure.

- 2.3.1, everything working. 
  - LAN/Wifi: ok Desktop, ok android, ok iphone 
  - 4G/Nabucasa: ok Android/iphone with 2x operators (TIM and HO Mobile)

- 3.x:
 - LAN/Wifi: ok Desktop, ok android, ok iphone (webrtc/MSE)
 - 4G/Nabucasa: ok Android/iphone with TIM (webrtc/MSE)
 - 4G/Nabucasa: NOK Android/iphone with HO Mobile (webrtc/MSE)

basically, with 3.x black screen with HO Mobile regardless the device or the protocol I try to use. 
without significant logs.

**Comment by Claudio1L** [comment]
of course, for me 2.3.1 is perfect. The only concern is about issues with future HA releases

**Comment by Dc1975** [comment]
I have the same problem, but without using Nabucasa 
HA Mobile App in my WLan => No Stream
HA Mobile Browser with WiFi => Stream
HA Mobile App with 4G => No Stream
HA Mobile Browser with 4G => Stream

The system configuration is:
 Home Assistant 2023.7.2
 Supervisor 2023.07.1
 Operating System 10.3
 Frontend 20230705.1 - latest 

The card configuration is:
type: custom:webrtc-camera
url: rtsp://user:pw@IP:554/h264Preview_01_sub
webrtc: true

I have changed webrtc true/false and url/entity in every condition with the same result described above.

![grafik](https://github.com/AlexxIT/WebRTC/assets/11534809/1ec88e93-a06c-4656-a07c-ffa9d74bafb1)
